### PR TITLE
Fix detection of AIOSEO

### PR DESCRIPTION
**Pull request by jdevalk**
Detection of the All In One SEO plugin was mostly broken for new versions.

This pull rectifies that and splits AIOSEO Pro out from free when possible.

**Test websites**:

- https://aioseo.com/
- https://wpbeginner.com/
- https://syedbalkhi.com/

### CODE REVIEW

File: src/technologies/a.json [hunk 1]
```diff
@@ -2454,21 +2454,53 @@
...
-    "description": "All in One SEO plugin optimizes WordPress website and its content for search engines.",
-    "html": "<!-- All in One SEO Pack ([\\d.]+) \\;version:\\1",
+    "description": "All in One SEO optimizes a WordPress website and its content for search engines.",
+    "meta": {
+      "generator": "^All in One SEO \\(AIOSEO\\) ([\\d\\.]+)\\;version:\\1"
+    },
+    "dom": {
+      "script.aioseo-schema": {
+        "attributes": {
+          "class": ""
```
Comment (tunetheweb) [inline src/technologies/a.json:2470]: Do we need these lines? And if it has `aioseo-schema` class then why is the class attribute empty?

File: src/technologies/a.json [hunk 2]
```diff
@@ -2454,21 +2454,53 @@
...
+    "html": "<!-- All in One SEO Pro ([\\d.]+) \\;version:\\1",
+    "meta": {
+      "generator": "^All in One SEO Pro \\(AIOSEO\\) ([\\d\\.]+)\\;version:\\1"
+    },
+     "icon": "AIOSEO.svg",
+    "pricing": [
+      "freemium",
+      "low",
+      "recurring"
+    ],
```
Comment (tunetheweb) [inline src/technologies/a.json:2502]: If this is the "Pro" option then are there freemium and low options? Not that we currentyl use this info, but still...

File: src/technologies/a.json [hunk 3]
```diff
@@ -2454,21 +2454,53 @@
...
+	"implies": "WordPress",
```
Comment (tunetheweb) [inline src/technologies/a.json:2503]: Is the indentation correct here?

File: src/technologies/a.json [hunk 4]
```diff
@@ -2454,21 +2454,53 @@
...
+    "cats": [
+      54,
+      87
+    ],
+    "description": "All in One SEO optimizes a WordPress website and its content for search engines.",
+    "html": "<!-- All in One SEO Pro ([\\d.]+) \\;version:\\1",
+    "meta": {
+      "generator": "^All in One SEO Pro \\(AIOSEO\\) ([\\d\\.]+)\\;version:\\1"
+    },
+     "icon": "AIOSEO.svg",
```
Comment (tunetheweb) [inline src/technologies/a.json:2497]: Indentation

```suggestion
    "icon": "AIOSEO.svg",
```

**Review by tunetheweb — CHANGES_REQUESTED** [review]
The test run doesn't seem to have picked up this technology with these changes. Not 100% sure why...

File: src/technologies/a.json (hunk 1 above)
Comment (jdevalk) [reply src/technologies/a.json:2470]: I honestly don't know, I copied this from Yoast SEO's detection :)

File: src/technologies/a.json (hunk 2 above)
Comment (jdevalk) [reply src/technologies/a.json:2502]: Also taken from how Yoast SEO is setup, but happy to change if I know what to change it to :P

File: src/technologies/a.json (hunk 3 above)
Comment (jdevalk) [reply src/technologies/a.json:2503]: no. I'll fix that.

File: src/technologies/a.json (hunk 2 above)
Comment (jdevalk) [reply src/technologies/a.json]: Changed it to how it's set for Yoast SEO entirely: `oss` to `true` for the free, and then the pricing thing on the premium thing.

**Comment by jdevalk** [comment]
@tunetheweb I think that's all of it? Let me know if I missed anything!

**Comment by tunetheweb** [comment]
@pmeenan @max-ostapenko any ideas why the detections are not being picked up by CI? The config looks correct to me but bit confused why it didn't pick this up. Is this a problem with the config, or with our testing checks?

**Comment by max-ostapenko** [comment]
Here is the fix - #36

**Comment by max-ostapenko** [comment]
@tunetheweb please restart the test action manually

**Comment by max-ostapenko** [comment]
Interesting - still nothing.
Will look into these particular changes.

File: src/technologies/a.json [hunk 5]
```diff
@@ -2454,21 +2454,45 @@
...
+    "description": "All in One SEO optimizes a WordPress website and its content for search engines.",
+    "meta": {
+      "generator": "^All in One SEO \\(AIOSEO\\) ([\\d\\.]+)\\;version:\\1"
+    },
+    "dom": {
+      "script.aioseo-schema": {}
+    },
+    "html": [
+      "<!-- All in One SEO Pack ([\\d.]+) \\;version:\\1",
+      "<!-- All in One SEO ([\\d.]+) \\;version:\\1"
```
Comment (max-ostapenko) [inline src/technologies/a.json:2472]: @jdevalk We could deduplicate the records by using the version attribute.

Example:
```suggestion
      "<!-- All in One SEO ([\\d.]+) \\;version:\\1",
      "<!-- All in One SEO Pro ([\\d.]+) \\;version:Pro \\1"
```

**Comment by max-ostapenko** [comment]
Update on test fix:

1. https://github.com/HTTPArchive/WebPageTest.api-nodejs/pull/4
we've forgot to merge the wappalyzer test feature into our node client

2. https://github.com/HTTPArchive/WebPageTest.api-nodejs/pull/5
fix for a parameter name is required for the node client too

3. https://github.com/HTTPArchive/wappalyzer/pull/38
dependencies updated with the latest version of the node client

**Comment by jdevalk** [comment]
Updated the pull request name, sorry about that :)

File: src/technologies/a.json (hunk 5 above)
Comment (jdevalk) [reply src/technologies/a.json:2472]: That looks smart!!

File: src/technologies/a.json [hunk 6]
```diff
@@ -2454,21 +2454,35 @@
...
     ],
     "cpe": "cpe:2.3:a:aioseo:all_in_one_seo:*:*:*:*:*:wordpress:*:*",
-    "description": "All in One SEO plugin optimizes WordPress website and its content for search engines.",
-    "html": "<!-- All in One SEO Pack ([\\d.]+) \\;version:\\1",
+    "description": "All in One SEO optimizes a WordPress website and its content for search engines.",
+    "meta": {
+      "generator": [
+		"^All in One SEO \\(AIOSEO\\) ([\\d\\.]+)\\;version:\\1",
+		"^All in One SEO Pro \\(AIOSEO\\) ([\\d\\.]+)\\;version:Pro \\1"
+	  ]
```
Comment (tunetheweb) [inline src/technologies/a.json:2468]: Nit: spacing.

```suggestion
		    "^All in One SEO \\(AIOSEO\\) ([\\d\\.]+)\\;version:\\1",
		    "^All in One SEO Pro \\(AIOSEO\\) ([\\d\\.]+)\\;version:Pro \\1"
	    ]
```

**Review by tunetheweb — APPROVED** [review]
LGTM

**Pull request merged**